resources/augeas.go

```diff
@@ -0,0 +1,249 @@
+// Mgmt
+// Copyright (C) 2013-2016+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"encoding/gob"
+	"fmt"
+	"honnef.co/go/augeas"
+	"log"
+	"strings"
+
+	"github.com/purpleidea/mgmt/event"
+	"github.com/purpleidea/mgmt/recwatch"
+
+	errwrap "github.com/pkg/errors"
+)
+
+func init() {
+	gob.Register(&AugeasRes{})
+}
+
+// AugeasRes is a resource that enables you to use the augeas resource.
+// Currently only allows you to change simple files (e.g sshd_config).
+type AugeasRes struct {
+	BaseRes    `yaml:",inline"`
+	File       string               `yaml:"file"` // the file targetted by this resource. If specified, mgmt will watch that file. Otherwhise, mgmt will poll the resource.
+	Lens       string               `yaml:"lens"` // the lens used by this resource. If specified, mgmt will watch that file.
+	Sets       [][]string           `yaml:"sets"` // A list of changes that will be applied to the file, in the form of ["path", "value"]. mgmt will run augeas.Get() before augeas.Set(), to prevent changing the file when it is not needed.
```

Comment:
The comments after struct members will get used automatically in the godoc which is great! For extremely long comments, I *think* you can have them right above the struct member in question (and on multiple lines) and they should still show up. Have a quick look into this if you can.